### ISSUE TITLE
Suggestion: place buildings on top of units

### ISSUE THREAD

**Issue by realPhint** [Sh0wdown/UnofficialCrusaderPatch#595]
Expacially on longer progressed games there are many workers (and drunks...) walking around, making the placing of buildings really annoying sometimes. As for the AI already does it, maybe it could be enabled to place buildings on top of units wich just dissapear then. Upon placing buildings on workers I guess they would simply respawn as peasents from the keep an be assigned anew while placed-on troops would be lost, for they cannot "respawn".
In case of towers and walls maybe the units could also be 'teleported' on them like at the AI.

**Comment by Admiral-Gumpert** [comment]
Overbuilding is my dream , As I always watch those AIs jealously do this all the time

**Comment by LordHansCapon** [comment]
Checked it out, and its turns out there are tons of check. Like whether your building requires Iron or Mars or Stone to be built on. Farms also check for whether they are on grass etc. All hardcoded with building IDs of course. However, there is a general collision testing with units, trees, rocks etc which I haven't located... must not be far though. But from the experience I have (which is aint much tbh), I bet its a bit harder to separate villagers from units or trees or rocks. May I be wrong... I always wanted it too.

**Comment by Truetobi91** [comment]
It might be important to also look for the consequences of building on top of units: What happens to a worker; does he respawn or not. I haven't found the issue but I also experienced that bug that if you place a building right in front of a tanner so that he will glitch onto it and despawn, he will not respawn unlike other workers. Will military units be deleted or teleported up on a tower (or just somewhere else see #470). Is the unit counter being updated correctly when deleting units? 

Side note: There's a problem just the other way around for the AI featuring the most buggy thing in Crusader: Engineers. The AI may constantly overbuild their freshly build harass siege engines because they didn't move fast enough.

**Comment by Krarilotus** [comment]
@Truetobi91 
> Side note: There's a problem just the other way around for the AI featuring the most buggy thing in Crusader: Engineers. The AI may constantly overbuild their freshly build harass siege engines because they didn't move fast enough.

I believe @patel-nikhil has already looked into the engineers beeing counted correctly, even after siege engines got destroyed. That way harassing siege engines now get build even later in the game i suppose.

**Comment by GRhin** [comment]
Yes, it is still being worked on, it was only found enough to be even
roughly sure it was possible just before the last release, wed have had to
delay release a month or more to include it. Hopefully next release. No
promises though.
